### ISSUE TITLE
Improper shut down

### ISSUE THREAD

**Issue by uRdoomd** [mbunkus/mkvtoolnix#2168]
When I'm using MKVToolNix GUI, and the app is left open then the computer is shut down/rebooted or the app is otherwise improperly closed, when it is reopened it will redo completed tasks. I don't know if it keeping the settings that were used for the completed task so I abort the job and redo it manually again. Running on windows 10 pro, build 1709 x64, did the same thing on 1703 x64. Has happened in previous version of the app as well, but currently using 18.0.0

To recreate it should be pretty straight forward
Complete a job
Leave app open
Reboot computer
Open app

**Comment by mbunkus** [comment]
This looks like a duplicate of #1787. The same comments apply here.

**Comment by uRdoomd** [comment]
This is NOT the same as the other post. That computer is shut down improperly. Mine ISN'T. It has nothing to do with write cache. The app can stay open for hours and then the computer is shut down or rebooted PROPERLY while the app is open... Time does not matter. Right away, or after hours of it being open. The comments on that post do NOT apply here and probably do not apply to the other post as well. It just sounds like an excuse not to look into it.

**Comment by mbunkus** [comment]
Ah. I misunderstood your description.

**Comment by uRdoomd** [comment]
Apologies for being rude. I just formatted my pc last week and probably should have saved some logs, but I do not have them anymore. I'm usually careful to close the app after all jobs have been completed but sometimes I don't notice it was left open and end up rebooting or shutting down. If there's any more information you need please let me know.

**Comment by mbunkus** [comment]
Don't worry about it. I don't think I'll need logs as I have some good ideas which portion of the source to look at. I haven't investigated yet, though, due to time constraints.

**Comment by mbunkus** [comment]
I haven't been able to reproduce this myself. Nevertheless, I've implemented a change that should prevent this behavior. Before that change the GUI only saved the job queue when the program exits. This is triggered even when the computer is shut down or rebooted. And therefore the queue _should_ be saved properly.

That being said, it's obvious it isn't in your case. Therefore the change: each job queue file is now saved whenever it changes its state to something that isn't "currently running". That should prevent behavior such as the one you've observed.

As I've said initially that I cannot reproduce the issue. Therefore I cannot really evaluate the effectiveness of said change. Can you please download the latest [debug build](https://mkvtoolnix.download/windows/debug/32-bit/) I've just uploaded and give it a try? Thanks.

**Comment by uRdoomd** [comment]
Thank you for taking the time to correct this. As I mentioned before, I had recently formatted my PC and so I cannot say for certain that your implemented fix that was the actual solution, or it may have been the format and fresh install of the app, however, I did test it and it absolutely 100% did not happen this time. Whether it was the format or your fix that did the trick, I appreciate it none the less. Again, Thank you.
